### PR TITLE
Activate GC on MP

### PR DESCRIPTION
**Pull request by MadsThy**
## Description

This PR is to allow for activating GC generation on metering points

Additional:

- Fixing the small errors that pop up around having used "end" for justify-content instead of "flex-end"
- Fixed the eo-popup-message to better show "see details" expandable link
- Made metering point table into a mat table, and enabled sorting in the table
- Created a working unit test for certificates service

### CODE REVIEW

File: libs/eo/metering-points/shell/src/lib/eo-metering-points.store.ts [hunk 1]
```diff
@@ -64,44 +56,71 @@ export class EoMeteringPointsStore extends ComponentStore<EoMeteringPointsState>
...
   );
 
-  readonly setError = this.updater(
+  private readonly setCertificateContract = this.updater(
+    (state, contract: EoCertificateContract): EoMeteringPointsState => ({
+      ...state,
+      meteringPoints: state.meteringPoints.map((mp) => {
+        if (mp.gsrn === contract.gsrn) {
+          mp.contract = contract;
+        }
```
Comment (duizer) [inline libs/eo/metering-points/shell/src/lib/eo-metering-points.store.ts:80]: At this point, this is fine as the contracts in the API only have a StartDate and not an EndDate, so please keep this as is. But in the future, EndDate will be added along with the possibility to cancel/stop a contract. This means, that it is possible to have multiple contracts per metering point.
FYI, the contracts returned by the API is only the contracts created by the signed-in user.

File: libs/eo/shared/atomic-design/feature-molecules/src/lib/eo-popup-message/eo-popup-message.component.ts [hunk 2]
```diff
@@ -108,30 +106,34 @@ import { MatIconModule } from '@angular/material/icon';
...
         </p>
-        <p class="inline">
+        <div style="display: flex; justify-content: space-between">
+          <p class="inline">
+            <a
+              href="mailto:datahub@energinet.dk?subject=Error on energioprindelse.dk&body=**Write your message here, but please keep the lines below intact for our internal investigations**%0D%0A%0D%0AError code:{{
+                errorMessage?.status
+              }}%0D%0A{{ errorMessage?.message }}"
+              >Contact customer service</a
+            >
```
Comment (duizer) [inline libs/eo/shared/atomic-design/feature-molecules/src/lib/eo-popup-message/eo-popup-message.component.ts:124]: Is this way of handling errors agreed with any of the POs (Søren or Andreas)?

**Review by duizer — CHANGES_REQUESTED** [review]

File: libs/eo/shared/atomic-design/feature-molecules/src/lib/eo-popup-message/eo-popup-message.component.ts (hunk 2 above)
Comment (duizer) [reply libs/eo/shared/atomic-design/feature-molecules/src/lib/eo-popup-message/eo-popup-message.component.ts:124]: Sorry, I missed that this is only presentation changes, not adding more to the error popup

File: libs/eo/metering-points/shell/src/lib/eo-metering-point-table.component.ts [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+      <ng-container matColumnDef="granular certificates">
+        <mat-header-cell *matHeaderCellDef mat-sort-header
+          >Granular Certificates</mat-header-cell
+        >
+        <mat-cell *matCellDef="let element">
+          <ng-container *ngIf="element.type === 'production'">
+            <span *ngIf="element.contract">Active</span>
+            <ng-container *ngIf="!element.contract">
+              <a
+                *ngIf="element.gsrn.length === 18"
```
[the remark this answers is not in the document]
Comment (MadsThy) [reply libs/eo/metering-points/shell/src/lib/eo-metering-point-table.component.ts:105]: Done :)

File: libs/eo/metering-points/shell/src/lib/eo-metering-point-table.component.ts [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+    this.dataSource.sortingDataAccessor = (item, property) => {
+      const itemHasActiveContract = item.contract ? 'active' : 'enable';
+
+      switch (property) {
+        case 'tags':
+          return item.type;
+        case 'address':
+          return item.address.address1.toLowerCase();
+        case 'granular certificates':
+          return item.type === 'production' && item.gsrn.length === 18
```
[the remark this answers is not in the document]
Comment (MadsThy) [reply libs/eo/metering-points/shell/src/lib/eo-metering-point-table.component.ts:157]: Fixed :)

File: libs/eo/metering-points/shell/src/lib/eo-metering-points.store.ts (hunk 1 above)
Comment (MadsThy) [reply libs/eo/metering-points/shell/src/lib/eo-metering-points.store.ts:80]: Thanks :) Good to know for the future 😄

**Comment by MadsThy** [comment]
@duizer Ready for rereview, so we can get it merged :)

**Review by duizer — APPROVED** [review]

**Review by duizer — APPROVED** [review]

**Review by duizer — APPROVED** [review]

**Pull request merged**